models/project_issue.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+
+	"xorm.io/xorm"
+)
+
+// ProjectIssue saves relation from issue to a project
+type ProjectIssue struct {
+	ID        int64 `xorm:"pk autoincr"`
+	IssueID   int64 `xorm:"INDEX"`
+	ProjectID int64 `xorm:"INDEX"`
+
+	// If 0, then it has not been added to a specific board in the project
+	ProjectBoardID int64 `xorm:"INDEX"`
+}
+
+func deleteProjectIssuesByProjectID(e Engine, projectID int64) error {
+	_, err := e.Where("project_id=?", projectID).Delete(&ProjectIssue{})
+	return err
+}
+
+//  ___
+// |_ _|___ ___ _   _  ___
+//  | |/ __/ __| | | |/ _ \
+//  | |\__ \__ \ |_| |  __/
+// |___|___/___/\__,_|\___|
+
+// LoadProject load the project the issue was assigned to
+func (i *Issue) LoadProject() (err error) {
+	return i.loadProject(x)
+}
+
+func (i *Issue) loadProject(e Engine) (err error) {
+	if i.Project == nil {
+		var p Project
+		if _, err = e.Table("project").
+			Join("INNER", "project_issue", "project.id=project_issue.project_id").
+			Where("project_issue.issue_id = ?", i.ID).
+			Get(&p); err != nil {
+			return err
+		}
+		i.Project = &p
+	}
+	return
+}
+
+// ProjectID return project id if issue was assigned to one
+func (i *Issue) ProjectID() int64 {
+	return i.projectID(x)
+}
+
+func (i *Issue) projectID(e Engine) int64 {
+	var ip ProjectIssue
+	has, err := e.Where("issue_id=?", i.ID).Get(&ip)
+	if err != nil || !has {
+		return 0
+	}
+	return ip.ProjectID
+}
+
+// ProjectBoardID return project board id if issue was assigned to one
+func (i *Issue) ProjectBoardID() int64 {
+	return i.projectBoardID(x)
+}
+
+func (i *Issue) projectBoardID(e Engine) int64 {
+	var ip ProjectIssue
+	has, err := e.Where("issue_id=?", i.ID).Get(&ip)
+	if err != nil || !has {
+		return 0
+	}
+	return ip.ProjectBoardID
+}
+
+//  ____            _           _
+// |  _ \ _ __ ___ (_) ___  ___| |_
+// | |_) | '__/ _ \| |/ _ \/ __| __|
+// |  __/| | | (_) | |  __/ (__| |_
+// |_|   |_|  \___// |\___|\___|\__|
+//               |__/
+
+// NumIssues return counter of all issues assigned to a project
+func (p *Project) NumIssues() int {
+	c, err := x.Table("project_issue").
+		Where("project_id=?", p.ID).
+		GroupBy("issue_id").Count("issue_id")
```

Comment:
as per @lunny xorm do not support this jet :(

Reply:
@lunny tested - this actualy works!!!

Reply:
Oh, no. It should not, you should not give `"issue_id"` to `Count`. Could you try below code?
```
x.Table("project_issue").
    Where("project_id=?", p.ID).
    GroupBy("issue_id").
    Cols("issue_id").
    Count()
```

Reply:
@lunny this works too - changed

Reply:
@lunny so `Count()` dont use the SQL COUNT(row) but count the rows itselve instead?